controller/src/main/java/io/pravega/controller/store/client/StoreClientFactory.java

```diff
@@ -115,14 +121,51 @@ public ZooKeeper newZooKeeper(String connectString, int sessionTimeout, Watcher
                 this.client = new ZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly);
                 return this.client;
             } else {
-                Preconditions.checkArgument(this.connectString.equals(connectString), "connectString differs");
-                Preconditions.checkArgument(this.sessionTimeout == sessionTimeout, "sessionTimeout differs");
-                Preconditions.checkArgument(this.canBeReadOnly == canBeReadOnly, "canBeReadOnly differs");
-                Preconditions.checkNotNull(watcher, "watcher");
-                this.client.register(watcher);
+                try {
+                    tryInjectFailure();
+                    Preconditions.checkArgument(this.connectString.equals(connectString), "connectString differs");
+                    Preconditions.checkArgument(this.sessionTimeout == sessionTimeout, "sessionTimeout differs");
+                    Preconditions.checkArgument(this.canBeReadOnly == canBeReadOnly, "canBeReadOnly differs");
+                    Preconditions.checkNotNull(watcher, "watcher");
+                    this.client.register(watcher);
+                } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
See the comment above, we don't need to catch NPE here.

Reply:
Done.